### PR TITLE
[SPARK-32888][DOCS] Add user document about header flag and RDD as path for reading CSV

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This proposes to enhance user document of the API for loading a Dataset of strings storing CSV rows. If the header option is set to true, the API will remove all lines same with the header.

### Why are the changes needed?

This behavior can confuse users. We should explicitly document it.

### Does this PR introduce _any_ user-facing change?

No. Only doc change.

### How was this patch tested?

Only doc change.

### CODE REVIEW

**Comment by viirya** [comment]
cc @HyukjinKwon

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -429,7 +429,8 @@ def csv(self, path, schema=None, sep=None, encoding=None, quote=None, escape=Non
         :param comment: sets a single character used for skipping lines beginning with this
                         character. By default (None), it is disabled.
         :param header: uses the first line as names of columns. If None is set, it uses the
-                       default value, ``false``.
+                       default value, ``false``. Note that if the given path is a RDD of Strings,
+                       this header option will remove all lines same with the header if exists.
```
Comment (HyukjinKwon) [inline python/pyspark/sql/readwriter.py:433]: Here too. I would use `.. note:` under this parameter.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 2]
```diff
@@ -600,6 +600,9 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * If the enforceSchema is set to `false`, only the CSV header in the first line is checked
    * to conform specified or inferred schema.
    *
+   * Note that if `header` option is set to `true` when calling this API, all lines same with
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala:603]: I would just `@note`

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise

**Comment by viirya** [comment]
Thanks @HyukjinKwon

**Comment by viirya** [comment]
Not sure why GitHub Actions failed at Python linter?

**Comment by HyukjinKwon** [comment]
I think .. Jenkins currently doesn't run PySpark documentation due to missing dependencies. I opened some JIRAs but looks like Shane is busy at this moment. GitHub Actions is the only build that build PySpark docs so that might be the reason.

**Comment by viirya** [comment]
Oh, got it. I'm fixing it.

**Comment by HyukjinKwon** [comment]
Merged to master, branch-3.0 and branch-2.4.

**Comment by viirya** [comment]
thanks @HyukjinKwon

**Pull request closed without merging**